### PR TITLE
Fix Windows-specific issues

### PR DESCRIPTION
**Pull request by collindutter**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

## Describe your changes
Fixes unit tests/core functionality on Windows machines. 
## Issue ticket number and link
NA

### CODE REVIEW

File: griptape/drivers/file_manager/amazon_s3_file_manager_driver.py
Comment (collindutter) [inline griptape/drivers/file_manager/amazon_s3_file_manager_driver.py:1]: Removed os-specific file system calls that happened to work on unix machines

**Comment by collindutter** [comment]
@dylanholmes will write up more detailed justifications for each change tomorrow

File: tests/unit/utils/test_file_utils.py [hunk 1]
```diff
@@ -14,7 +14,6 @@ def test_load_file(self):
         file = utils.load_file(os.path.join(dirname, "../../resources/foobar-many.txt"))
 
         assert file.decode("utf-8").startswith("foobar foobar foobar")
-        assert len(file.decode("utf-8")) == 4563
```
Comment (dylanholmes) [inline tests/unit/utils/test_file_utils.py:17]: I think this is the only change that surprised me. Is the txt file size os dependent? Is this related to different line endings or something?

**Review by dylanholmes — COMMENTED** [review]
Nice work!

Only one question, everything else makes sense

File: tests/unit/utils/test_file_utils.py (hunk 1 above)
Comment (collindutter) [reply tests/unit/utils/test_file_utils.py:17]: Yeah must be from line endings -- it was slightly different per OS.

**Comment by collindutter** [comment]
Great 👍

Gonna add a windows runner in GitHub actions and then will mark as ready.

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -24,3 +24,23 @@ jobs:
         uses: ./.github/actions/init-environment 
       - name: Run unit tests
         run: make test/unit
+  test-windows:
```
Comment (dylanholmes) [inline .github/workflows/unit-tests.yml:27]: THANK YOU!

**Review by dylanholmes — APPROVED** [review]
So awesome that you added the windows runner in this PR
